Parser/tokenizer.c

```diff
@@ -1075,6 +1080,12 @@ tok_nextc(struct tok_state *tok)
             return EOF;
         }
         tok->line_start = tok->cur;
+
+        if (contains_null_bytes(tok->line_start, tok->inp - tok->line_start)) {
+            syntaxerror(tok, "source code string cannot contain null bytes");
```

Comment:
We get here when reading from a file too, right?
```suggestion
            syntaxerror(tok, "source code cannot contain null bytes");
```

Reply:
Done!